ontology/rml-core.owl

```diff
@@ -0,0 +1,517 @@
+               rdfs:comment "Represents the logical target to where triples are exported to. This can be a pointer to any dataset." ;
+               rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+               rdfs:label "logical target" .
+
+
+###  http://w3id.org/rml/object
+:object rdf:type owl:ObjectProperty ;
+        rdfs:domain :PredicateObjectMap ;
+        rdfs:range rdfs:Resource ;
+        rdfs:comment "Specifies the object for the generated triple from the logical source record."@en ;
+        rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+        rdfs:label "object" .
+
+
+###  http://w3id.org/rml/objectMap
+:objectMap rdf:type owl:ObjectProperty ;
+           rdfs:domain :PredicateObjectMap ;
+           rdfs:range [ rdf:type owl:Class ;
+                        owl:unionOf ( :ObjectMap
+                                      :RefObjectMap
+                                      :StarMap
+                                    )
+                      ] ;
+           rdfs:comment "An ObjectMap element to generate the object component of the (predicate, object) pair from a logical source record."@en ;
+           rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+           rdfs:label "object map" .
+
+
+###  http://w3id.org/rml/parentTriplesMap
+:parentTriplesMap rdf:type owl:ObjectProperty ;
+                  rdfs:domain :RefObjectMap ;
+                  rdfs:range :TriplesMap ;
+                  rdfs:comment "Specifies the TriplesMap element corresponding to the parent logical table of the foreign key constraint."@en ;
+                  rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+                  rdfs:label "parent triples map" .
+
+
+###  http://w3id.org/rml/predicate
+:predicate rdf:type owl:ObjectProperty ;
+           rdfs:domain :PredicateObjectMap ;
+           rdfs:range rdf:Property ;
+           rdfs:comment "Specifies the predicate for the generated triple from the logical source record."@en ;
+           rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+           rdfs:label "predicate" .
+
+
+###  http://w3id.org/rml/predicateMap
+:predicateMap rdf:type owl:ObjectProperty ;
+              rdfs:domain :PredicateObjectMap ;
+              rdfs:range :PredicateMap ;
+              rdfs:comment "A PredicateMap element to generate the predicate component of the (predicate, object) pair from a logical source record."@en ;
+              rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+              rdfs:label "predicate map" .
+
+
+###  http://w3id.org/rml/predicateObjectMap
+:predicateObjectMap rdf:type owl:ObjectProperty ;
+                    rdfs:range :PredicateObjectMap ;
+                    rdfs:comment "A PredicateObjectMap element to generate (predicate, object) pair from a logical source record."@en ;
+                    rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+                    rdfs:label "predicate object map" .
+
+
+###  http://w3id.org/rml/subject
+:subject rdf:type owl:ObjectProperty ;
```

Comment:
The object of :subject might be a class, depending on the use case. Whereas owl:ObjectProperty restricts to an instance, right?

Reply:
I believe `owl:ObjectProperty` is correct here. A class is also an instance, namely of `owl:Class`.

Reply:
Right, in OWL full though.
Do we specify the flavor of OWL in which the ontology is to be treated?